framework/include/utils/TransitionLayerTools.h

```diff
@@ -0,0 +1,210 @@
+ * @param end_side_boundary_id id of the generated mesh's external boundary that connects the last
+ * Points of the two input Point vectors
+ * @param type type of the MOOSE object that calls this method for error messages
+ * @param name name of the MOOSE object that calls this method for error messages
+ * @param quad_elem whether the QUAD4 elements are used to construct the mesh
+ * @param bias_parameter a parameter to control bias options (1) >0, biasing factor (2) <=0
+ * automatic biasing
+ * @param sigma Gaussian parameter used for Gaussian blurring of local node density; only used if
+ * bias_parameter <= 0
+ */
+void transitionLayerGenerator(ReplicatedMesh & mesh,
+                              const std::vector<Point> boundary_points_vec_1,
+                              std::vector<Point> boundary_points_vec_2,
+                              const unsigned int num_layers,
+                              const subdomain_id_type transition_layer_id,
+                              const boundary_id_type input_boundary_1_id,
+                              const boundary_id_type input_boundary_2_id,
+                              const boundary_id_type begin_side_boundary_id,
+                              const boundary_id_type end_side_boundary_id,
+                              const std::string type,
+                              const std::string name,
+                              const bool quad_elem = false,
+                              const Real bias_parameter = 1.0,
+                              const Real sigma = 3.0);
+
+/**
+ * Generates a 2D mesh with triangular elements for a region defined by two curves (sets of Points)
+ * @param mesh a reference ReplicatedMesh to contain the generated mesh
+ * @param boundary_points_vec_1 vector of Points of Side #1
+ * @param boundary_points_vec_2 vector of Points of Side #2
+ * @param num_layers number of sublayers of elements
+ * @param transition_layer_id subdomain id of the generated mesh
+ * @param external_boundary_id id of the generated mesh's external boundary
+ * @param type type of the MOOSE object that calls this method
+ * @param name name of the MOOSE object that calls this method
+ * @param quad_elem whether the QUAD4 elements are used to construct the mesh
+ */
+void transitionLayerGenerator(ReplicatedMesh & mesh,
+                              const std::vector<Point> boundary_points_vec_1,
+                              const std::vector<Point> boundary_points_vec_2,
+                              const unsigned int num_layers,
+                              const subdomain_id_type transition_layer_id,
+                              const boundary_id_type external_boundary_id,
+                              const std::string type,
+                              const std::string name,
+                              const bool quad_elem = false);
+
+/**
+ * Generates a 2D mesh based on a 2D vector of Nodes using QUAD4 elements
```

Comment:
this seems like a very general capability, but it s a little more specific right? It only works with a certain positioning of the nodes right?

Reply:
The nodes need to be on XY Plane for now. Will add this to the comments.